aws/request/request_1_8.go

```diff
@@ -9,3 +9,20 @@ import (
 // NoBody is a http.NoBody reader instructing Go HTTP client to not include
 // and body in the HTTP request.
 var NoBody = http.NoBody
+
+// ResetBody rewinds the request body back to its starting position, and
+// set's the HTTP Request body reference. When the body is read prior
+// to being sent in the HTTP request it will need to be rewound.
+//
+// Will also set the Go 1.8's http.Request.GetBody member to allow retrying
+// PUT/POST redirects.
+func (r *Request) ResetBody() {
```

Comment:
The one thing I think that is important here to mention is that ResetBody must be called before calling sending this over the wire, because if not, then the GetBody would never get set and redirects would not work.